### ISSUE TITLE
Docs: restrict-plus-operands example is wrong for the default configuration

### ISSUE THREAD

**Issue by lgenzelis** [typescript-eslint/typescript-eslint#7297]
### Before You File a Documentation Request Please Confirm You Have Done The Following...

- [X] I have looked for existing [open or closed documentation requests](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Adocumentation) that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Suggested Changes

[restrict-plus-operands docs](https://typescript-eslint.io/rules/restrict-plus-operands/) starts by giving these incorrect examples:

```typescript
let foo = '5.5' + 5;
```

I remember that was once the case, and I don't know when it changed, but now that's not an error anymore. The default configuration is:

```javascript
const defaultOptions: Options = [
  {
    allowAny: true,
    allowBoolean: true,
    allowNullish: true,
    allowNumberAndString: true,
    allowRegExp: true,
    skipCompoundAssignments: false,
  },
];
```

Since we have `allowNumberAndString: true`, it's ok to do `'5.5' + 5`. It can be verified [in the playground](https://typescript-eslint.io/play/#ts=5.1.6&fileType=.ts&code=DYUwLgBAZg9jEF4IHICsA6VyIGoKoG4AoIA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6aFfKsu4%2BWZLQD2paAEMmAE1QZIiaNCHRI4MAF8QaoA&tsconfig=&tokens=false).

### Affected URL(s)

https://typescript-eslint.io/rules/restrict-plus-operands/

**Comment by BrianLeishman** [comment]
Also running into this, and it's quite confusing. Shouldn't all of these really default to `false` instead of `true`?

The `Caution` note seems to imply that these *shouldn't* be set to true at all.

**Comment by ad1992** [comment]
Looks like this issue was fixed by #8362  so can be closed?

**Comment by auvred** [comment]
Yes, thanks!
